oshmem/mca/atomic/atomic.h

```diff
@@ -86,17 +86,25 @@ struct mca_atomic_base_module_1_0_0_t {
     opal_object_t super;
 
     /* Collective function pointers */
+    int (*atomic_add)(void *target,
+                      const void *value,
+                      size_t size,
+                      int pe);
     int (*atomic_fadd)(void *target,
                        void *prev,
                        const void *value,
-                       size_t nlong,
-                       int pe,
-                       struct oshmem_op_t *op);
+                       size_t size,
```

Comment:
how about making "prev" a "uint64_t*"? will avoid some branches

Reply:
not sure I got your idea...
for now UCX implementation operates by prev value directly, without any casting to datatype or branches

Reply:
for example in cswap:
```
 if (4 == size) {
            *(uint32_t*)prev = val;
        } else {
            *(uint64_t*)prev = val;
```

Reply:
this is not "user data", it's a local variable created in the shmem wrapper, which can always be 64-bit
check https://www.open-mpi.org/doc/v3.0/man3/shmem_long_fadd.3.php - user does not pass pointer to prev, the prev value is returned from the API call

Reply:
got it
done